SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialStrategySystem.cpp

```diff
@@ -1,58 +1,306 @@
+			{
+				const SpatialGDK::ComponentData* IntentData = Element.Components.FindByPredicate(
+					SpatialGDK::ComponentIdEquality{ SpatialConstants::AUTHORITY_INTENTV2_COMPONENT_ID });
+				check(IntentData);
+				AuthorityIntentView.Add(Delta.EntityId, AuthorityIntentV2(IntentData->GetUnderlying()));
+			}
+
+			{
+				const SpatialGDK::ComponentData* DelegationData = Element.Components.FindByPredicate(
+					SpatialGDK::ComponentIdEquality{ SpatialConstants::AUTHORITY_DELEGATION_COMPONENT_ID });
+				check(DelegationData);
+				AuthorityDelegationView.Add(Delta.EntityId, AuthorityDelegation(DelegationData->GetUnderlying()));
+			}
+
+			{
+				const SpatialGDK::ComponentData* NetOwnerData = Element.Components.FindByPredicate(
+					SpatialGDK::ComponentIdEquality{ SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID });
+				if (ensureAlwaysMsgf(NetOwnerData, TEXT("Entity %llu missing net owner data"), Delta.EntityId))
+				{
+					auto& NetOwner = NetOwningClientView.Add(Delta.EntityId, NetOwningClientWorker(NetOwnerData->GetUnderlying()));
+					if (NetOwner.ClientPartitionId.IsSet())
+					{
+						EntitiesClientChanged.Add(Delta.EntityId);
+					}
+				}
+			}
+
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnAdded(Delta.EntityId, Element);
+			}
 		}
 		break;
 		case EntityDelta::REMOVE:
+		{
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnRemoved(Delta.EntityId);
+			}
+			AuthorityIntentView.Remove(Delta.EntityId);
+			PendingMigrations.Remove(Delta.EntityId);
+		}
+		break;
 		case EntityDelta::TEMPORARILY_REMOVED:
 		{
-			// TODO
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnRemoved(Delta.EntityId);
+			}
+			const SpatialGDK::EntityViewElement& Element = LBView.GetView().FindChecked(Delta.EntityId);
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnAdded(Delta.EntityId, Element);
+			}
 		}
 		break;
 		default:
 			break;
 		}
 	}
+
+	Strategy->Advance(Connection);
 }
 
-void SpatialStrategySystem::Flush(SpatialOSWorkerInterface* Connection)
+void FSpatialStrategySystem::Flush(ISpatialOSWorker& Connection)
 {
-	// TODO
+	PartitionsMgr->Flush(Connection);
+
+	if (bStrategySystemInterestDirty)
+	{
+		UpdateStrategySystemInterest(Connection);
+	}
+
+	Strategy->TickPartitions(*PartitionsMgr);
+	Strategy->Flush(Connection);
+
+	TSet<Worker_EntityId_Key> ModifiedEntities;
+	for (auto& Storage : DataStorages)
+	{
+		ModifiedEntities = ModifiedEntities.Union(Storage->GetModifiedEntities());
+	}
+
+	FMigrationContext Ctx(MigratingEntities, ModifiedEntities);
+
+	Strategy->CollectEntitiesToMigrate(Ctx);
+
+	for (auto& Storage : DataStorages)
+	{
+		Storage->ClearModified();
+	}
+
+	for (auto PendingMigration : PendingMigrations)
+	{
+		if (!Ctx.EntitiesToMigrate.Contains(PendingMigration.Key))
+		{
+			Ctx.EntitiesToMigrate.Add(PendingMigration);
+		}
+	}
+	PendingMigrations.Empty();
+
+	for (auto const& Migration : Ctx.EntitiesToMigrate)
```

Comment:
I don't see where `EntitiesToMigrate` is cleared

Reply:
FMigrationContext Ctx is a stack variable.